frontend/src/components/object/ObjectAccess.vue

```diff
@@ -1,12 +1,41 @@
-<!-- <script setup lang="ts">
+<script setup lang="ts">
+import { onMounted, ref, watch} from 'vue';
+import { storeToRefs } from 'pinia';
+import { useUserStore } from '@/store';
+import { Permissions } from '@/utils/constants';
 
-defineProps({
+import type { Ref } from 'vue';
+import type { Permission } from '@/interfaces';
+
+const userStore = useUserStore();
+const { userSearch } = storeToRefs(userStore);
+
+const props = defineProps({
   objectAccess: {
     type: Object,
     default: undefined
   }
 });
 
+const managedBy: Ref<string | undefined> = ref();
+
+async function getUserData() {
+  const uniqueIds = [...new Set(props.objectAccess?.permissions
+    .filter( (x: Permission) => x.permCode === Permissions.MANAGE )
+    .map((x: Permission) => x.userId))];
+
+  await userStore.searchUsers({userId: uniqueIds} );
+
+  managedBy.value = userSearch.value.map( x => x.fullName ).join( ', ');
+}
+
+onMounted(() => {
+  getUserData();
+});
+
+watch( props, async () => {
+  await getUserData();
+});
```

Comment:
If I understand this correctly, this registers an event hook handler when props changes (assuming it detects deep object reassignments). While the async and await thing does make sense, in this case, it may be better to omit that as events are already asynchronously handled by nature of being triggered by an event. There's also nothing in the code that needs to explicitly block on the completion of getUserData, so the await (and async annotation on the callback) isn't actually needed here.